### PR TITLE
Add support for --autoMerge and --autoMergeMethod

### PR DESCRIPTION
**Pull request by brianseeders**
Implements support for setting auto-merge on created backport PRs.

See
https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request
and
https://docs.github.com/en/graphql/reference/mutations#enablepullrequestautomerge

### CODE REVIEW

**Comment by brianseeders** [comment]
@sqren I did implement a `enablePullRequestAutoMerge.private.test.ts` pointed to my fork for testing, but didn't commit it. I wasn't sure how you would want to handle.

**Comment by sqren** [comment]
> I did implement a enablePullRequestAutoMerge.private.test.ts pointed to my fork for testing, but didn't commit it. I wasn't sure how you would want to handle.

Awesome! Can you commit it and I'll create a repo under https://github.com/backport-org.
Btw. lmk if you have better ideas for these types of e2e tests. I've found them super valuable and easy to write but it seems a bit fishy that I have to create the repo like in this case.

**Comment by brianseeders** [comment]
I can, but I'll have to flesh it out more. For example, it needs to make
sure that the option gets disabled after each test

File: src/services/github/v4/enablePullRequestAutoMerge.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      repoOwner,
+      repoName,
+      pullNumber,
+    },
+  });
+
+  const pullRequestId = prResponse.repository.pullRequest.id;
+
+  const query = /* GraphQL */ `
+    mutation EnablePullRequestAutoMerge(
```
Comment (sqren) [inline src/services/github/v4/enablePullRequestAutoMerge.ts:56]: Nice! I never tried mutations before. Always feel back to v3 api for those operations. This looks great though.

**Comment by sqren** [comment]
> For example, it needs to make sure that the option gets disabled after each test

Ah, true!

File: src/services/github/v4/enablePullRequestAutoMerge.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+  `;
+
+  const res = await apiRequestV4<PullRequestAutoMergeResponse>({
+    githubApiBaseUrlV4,
+    accessToken,
+    query,
+    variables: {
+      pullRequestId,
+      mergeMethod: (autoMergeMethod || 'merge').toUpperCase(),
```
Comment (sqren) [inline src/services/github/v4/enablePullRequestAutoMerge.ts:76]: You may want to add the default to: https://github.com/sqren/backport/blob/4dcf0ea32e988df1f2fb2b7ac3e1b0c5187c8836/src/options/config/config.ts#L13

File: src/services/github/v4/enablePullRequestAutoMerge.ts (hunk 1 above)
Comment (brianseeders) [reply src/services/github/v4/enablePullRequestAutoMerge.ts:56]: Same here, I've never even used GraphQL before. This option isn't available in the REST api though

Comment (sqren) [reply src/services/github/v4/enablePullRequestAutoMerge.ts:56]: Do you use https://github.com/graphql/graphiql ? I find it invaluable for exploring the API and trying out different queries.

Comment (brianseeders) [reply src/services/github/v4/enablePullRequestAutoMerge.ts:56]: Yeah, I used GitHub's hosted version of it: https://docs.github.com/en/graphql/overview/explorer

**Comment by brianseeders** [comment]
@sqren I added `enablePullRequestAutoMerge.private.test.ts`. The tests are currently gated until the `TEST_*` constants at the top are filled in.

> lmk if you have better ideas for these types of e2e tests.

When you're writing something that's whole purpose is to integrate with a thing, and you can't run your own version of that thing, this is pretty much all you can do for e2e tests. It's super easy to just say "it's not worth it" and not do e2e tests at all, so good on you for setting them up!

**Comment by sqren** [comment]
Released in v5.6.5

**Pull request merged**